### PR TITLE
Support for updates to ACP

### PR DESCRIPTION
**Pull request by edwardsph**
This is a significant change. The series of commits represent the following:
* Move existing ACP support to *AcpLegacy* classes
* Implement changes to the *Acp* classes to use acp:Matcher instead of acp:Rule and switch to acl for access modes
* Detect the version of ACP being used based on a feature in the test subject description
* There was an early push (3 commits) along with some debugging tests for the ESS team to test with
* Next the test fixture creation and ACL modifications are added to the Karate logs to appear in reports
* The remaining 8 commits were cleaning up, bug fixing and improving unit tests

### CODE REVIEW

File: example/web-access-control/protected-operation/acl-propagation.feature [hunk 1]
```diff
@@ -40,20 +39,16 @@ Feature: Inheritable ACL controls child resources
     And headers clients.bob.getAuthHeaders('PUT', resource2.getUrl())
     And header Content-Type = 'text/plain'
     When method PUT
-    Then status 201
+    Then match [200, 201, 204, 205] contains responseStatus
```
Comment (kjetilk) [inline example/web-access-control/protected-operation/acl-propagation.feature:42]: Actually, that test is correct, only `201` is allowed by HTTP when a resource is created.

File: example/web-access-control/protected-operation/debug-direct-acl.feature [hunk 2]
```diff
@@ -0,0 +1,33 @@
+Feature: Test ACL creation in apply mode
```
Comment (kjetilk) [inline example/web-access-control/protected-operation/debug-direct-acl.feature:1]: OK, I have to admit that I haven't look at the actual ACR spec, as the draft hasn't yet been merged.

File: src/test/java/TestSuiteRunner.java [hunk 3]
```diff
@@ -59,7 +59,12 @@ void setup() {
     void testSuite() throws IOException {
         conformanceTestHarness.initialize();
         final List<String> filters = null;
-//        final List<String> filters = List.of("turtle");
+//        final List<String> filters = List.of("containment");
+//        final List<String> filters = List.of("protected");
+//        final List<String> filters = List.of("web-access-control");
+//        final List<String> filters = List.of("container");
+//        final List<String> filters = List.of("debug-direct", "debug-indirect", "retro");
+//        final List<String> filters = List.of("containment", "content", "container", "read-resource-default-R");
```
Comment (kjetilk) [inline src/test/java/TestSuiteRunner.java:67]: Why do you leave these comments in?

**Review by kjetilk — CHANGES_REQUESTED** [review]
It looks mostly good to me. There's one change I think that should be reverted, a test that looked correct to me.

But I would not that I don't know the code in sufficient detail to see if it is best practice or to spot a bug if it existed, but I'll leave it up to you to decide if you want another pair of eyes to look :-)

File: example/web-access-control/protected-operation/debug-indirect-acl.feature [hunk 4]
```diff
@@ -0,0 +1,44 @@
+Feature: Test ACL in applyMembers mode
```
Comment (acoburn) [inline example/web-access-control/protected-operation/debug-indirect-acl.feature:1]: The current ACR draft does not use applyMembers. Instead, it uses memberAccessControl

https://solid.github.io/authorization-panel/acp-specification/#inheritance

File: src/main/java/org/solid/testharness/accesscontrol/AccessDatasetAcp.java [hunk 5]
```diff
@@ -60,28 +62,34 @@
...
                 i.addAndGet(1);
                 final List<String> accessModes = rule.getAccess().stream()
                         .filter(mode -> !controlModes.contains(mode))
                         .collect(Collectors.toList());
                 final List<String> controlAccessModes = extractControlModes(rule);
+                final IRI accessControlNode = iri(namespace, "accessControl" + i);
                 final IRI policyNode = iri(namespace, "policy" + i);
                 final IRI ruleNode = iri(namespace, "rule" + i);
-                builder.subject(iri(aclUri))
+                builder.add(accessControlResource, ACP.accessControl, accessControlNode)
```
Comment (acoburn) [inline src/main/java/org/solid/testharness/accesscontrol/AccessDatasetAcp.java:86]: In the current draft, inheritance happens at the AccessControl level, not the Policy level

File: example/web-access-control/protected-operation/debug-indirect-acl.feature (hunk 4 above)
Comment (edwardsph) [reply example/web-access-control/protected-operation/debug-indirect-acl.feature:1]: As discussed, the final changes to ACR will be supported in a separate update once they are implemented such that the harness can be tested.

File: example/web-access-control/protected-operation/acl-propagation.feature (hunk 1 above)
Comment (edwardsph) [reply example/web-access-control/protected-operation/acl-propagation.feature:42]: The same change was incorrectly made to the specification-tests version recently so I will revert that too and restore the comment that explained why 201 was mandated for a created resource

File: example/web-access-control/protected-operation/debug-direct-acl.feature (hunk 2 above)
Comment (edwardsph) [reply example/web-access-control/protected-operation/debug-direct-acl.feature:1]: Both spec and implementations are evolving so there will be more changes to come

**Pull request merged**